uvm/api/com/untangle/uvm/PasswordUtil.java

```diff
@@ -19,6 +22,71 @@ public class PasswordUtil
     // We pass no seed here, so we'll get different random numbers each time.
     private static SecureRandom srng = new SecureRandom();
 
+    private final static Logger logger = LogManager.getLogger(PasswordUtil.class);
+
+    private final static String passwordEncryptionCmd = "/usr/bin/password-manager -e ";
+    private final static String passwordDecryptionCmd = "/usr/bin/password-manager -d ";
+    
+    
+        /**  ************************* NOTE *************************
+         *  Currently binary is generating warn related to TPM including the password in second line
+         *  so handled the code accordingly in getEncryptPassword() and getDecryptPassword()
+         *  Also while executing the command password is shown in log, I  handled this by overriding rhe execOutput() 
+         *  and sending the value as false so you will not see any cmd log for password encryption and decryption
+         * 
+         * To get it in log for debug purpose you can set the value as true
+         * 
+        */
+        /**
+         * Encrypt the provided password by invoking an password manager
+         * command to retrieve the encrypted value.
+         * 
+         * @param password a <code>String</code> containing the password to be encrypt
+         * @return a <code>String</code> containing the encrypt password
+        */
+        public static String getEncryptPassword(String password){
+            try {
+                String command = passwordEncryptionCmd + password;
+                String cmdOutput =  UvmContextFactory.context().execManager().execOutput(false, command);
+                String[] encryptPassword = cmdOutput.split("\n");
+                if (encryptPassword.length <= 1) {
+                    throw new IllegalStateException("Encryption output is invalid.");
+                }
+                return encryptPassword[1];               
+            } catch (IllegalStateException e) {
+                logger.info("Encryption output is invalid."+ e);
+            } 
+            catch (Exception e) {
+                logger.info("Exception occured while encrypting the password"+ e);
+            }
+            return null;
```

Comment:
Caller may not do null check so throwing an exception is also a good idea but your choice.

Reply:
This approach was implemented to handle exceptions in one central location, so the caller only needs to check for a null value instead of handling different exceptions separately.